app/js/filter-by-file-targets-directive.js

```diff
@@ -0,0 +1,182 @@
+/* Directives */
+angular.module('cttvDirectives')
+
+/**
+ * Options for configuration are:
+ * Example:
+ *   <cttv-filter-by-file-targets> </cttv-filter-by-file-targets>
+ *
+ */
+    .directive('cttvFilterByFileTargets', ['$log','cttvAPIservice', 'cttvFiltersService','$q',function($log, cttvAPIservice, cttvFiltersService, $q){
+        'use strict';
+
+
+
+        //$log.log("cttvFilterByFileTargets");
+
+        return {
+
+            //restrict:'E',
+            scope:true,
+            //template: 'Hello WORLD'
+            templateUrl: 'partials/filter-by-file-targets.html',
+            link: function(scope, elem, attrs){
+                //$log.log("cttvFilterByFileTargets:linkFunction: scope", scope);
+                //$log.log("cttvFilterByFileTargets:linkFunction: elem", elem);
+
+                scope.initFilterByFile =function(){
+
+                    scope.fileName = "";
+
```

Comment:
Line 31 - here is the part that I do not like - directive should not know where it is being used.